data/test/scenarios/wml_tests/ScenarioWML/EventWML/events-test_filter_ability.cfg

```diff
@@ -226,6 +226,76 @@
     [/event]
 )}
 
+##
+# Actions:
+# Use the common setup from FILTER_ABILITY_TEST.
+# Add an event with a filter_wml matching Alice's drains ability.
+# Alice attacks Bob and then Bob attacks Alice, as defined in FILTER_ABILITY_TEST.
+##
+# Expected end state:
+# The filtered event is triggered exactly once.
+#####
+{GENERIC_UNIT_TEST event_test_filter_ability_wml (
+    {FILTER_ABILITY_TEST}
+    [event]
+        name=attack
+        first_time_only=no
+        [filter]
+            [experimental_filter_ability]
+                tag_name=drains
+                [filter_wml]
+                    [filter]
+                        [filter_location]
+                            time_of_day=neutral
+                        [/filter_location]
+                    [/filter]
+                [/filter_wml]
+            [/experimental_filter_ability]
+        [/filter]
+        {ASSERT ({VARIABLE_CONDITIONAL side_number equals 1})}
+        {ASSERT ({VARIABLE_CONDITIONAL triggers equals 0})}
+        {VARIABLE_OP triggers add 1}
+    [/event]
+    [event]
+        name=turn 2
```

Comment:
If you're checking that it's triggered exactly once, this needs `first_time_only=no`